astropy/coordinates/builtin_frames/icrs_cirs_transforms.py

```diff
@@ -44,9 +43,12 @@ def icrs_to_cirs(icrs_coo, cirs_frame):
         # astrometric coordinate direction and *then* run the ERFA transform for
         # no parallax/PM. This ensures reversiblity and is more sensible for
         # inside solar system objects
-        newxyz = icrs_coo.cartesian.xyz.T - astrom['eb']*u.au
-        newcart = CartesianRepresentation(newxyz.T)
-
+        newxyz = intermedrep.to_cartesian().xyz
+        newxyz = np.rollaxis(newxyz,-1,0) -  astrom['eb']*u.au
```

Comment:
I think these `rollaxis` would still only work for 2D arrays of shape `(3, n)`, but one wants them to work for more complex shapes `(3, n1, n2, n3)`. Since `astrom` has shape `(..., 3)`, I think we need to place the first axis of shape `3` to the last place:
```
newxyz = np.rollaxis(newxyz, 0, newxyz.ndim) - astrom['eb'] * u.au
```
The next line then correctly rolls back the last axis to the front (aside: PEP8 wants spaces after `,`).